net/local/local_conn.c

```diff
@@ -128,6 +128,16 @@ FAR struct local_conn_s *local_alloc(void)
       nxsem_init(&conn->lc_donesem, 0, 0);
       nxsem_set_protocol(&conn->lc_donesem, SEM_PRIO_NONE);
 
+#endif
+
+#ifdef CONFIG_NET_LOCAL_DGRAM
```

Comment:
why check CONFIG_NET_LOCAL_DGRAM

Reply:
I only add lc_sendsem for udp. So if CONFIG_NET_LOCAL_DGRAM is not enabled, the sem does not exist

Reply:
As I said TCP has the similar issue regardless whether the data is sent in one or multiple file_write.

Reply:
okay, will fix that